Homepage redesign

coryjewell opened the pull request:
### Why
- Homepage still hasn't been touched with the UI refresh

### How
- Rebuilt homepage and global footer, mobile styling all completed
- Added **_light_** fade in animations using https://github.com/mamunhpath/josh.js
- Other: removed Health from the devices index table and put a health indicator in the ID column (will only appear when a device is quarantined and you can still filter by health).

jjcarstens commented:
Can you rebase this against `main`? It looks like it has a few comments from previous deployment changes?

fhunleth commented:
Also, please separate out the changes not involved with the home page into separate PRs.

coryjewell commented:
Reverted the health changes and will open a new PR. 

Fixed the JS test, the elixir test that was failing is also failing on `main` and I'm not sure about the cause.

I did rebase from main but made a mistake on the conflict resolution on a few lines of css. I cleaned the code up (just duplicate css) but the commits are still there.

jjcarstens commented:
This should fix the failing test. Not sure how it got in there...
https://github.com/nerves-hub/nerves_hub_web/pull/647

fhunleth commented on apps/nerves_hub_www/lib/nerves_hub_www_web/templates/account/invite.html.eex:
```diff
@@ -6,7 +6,7 @@
   <%= form_for @changeset, Routes.account_path(@conn, :accept_invite, @token), [as: :user, method: "post", class: "form-page"], fn f -> %>
     <div class="form-group">
-      <label for="username_input">Email/Username</label>
+      <label for="username_input">Username</label>
```
This change doesn't look like part of the homepage redesign. If there are issues holding up the redesign, I think this one could be broken out and merged quickly.

fhunleth commented on apps/nerves_hub_www/lib/nerves_hub_www_web/templates/home/index.html.eex:
```diff
@@ -1,39 +1,148 @@
+    <img src="/images/home/send-updates.png" alt="Delta updates" />
+    <div class="screen-description josh-js" data-josh-anim-name="fadeIn">
+      <h4>Send Smaller Files with Delta Updates</h4>
+      <p>Reduce the size — and cost — of your over-the-air firmware updates by up to 95%. Delta updates compare two binary files and only update the pieces of the firmware that changed, consuming less bandwidth every time.</p>
```
Delta updates are huge, but it's not ready yet. Maybe this should be marked "Coming soon?"

coryjewell replied:
Delta updates have already been merged into main!

https://github.com/nerves-hub/nerves_hub_web/pull/627
https://github.com/nerves-hub/nerves_hub_web/pull/648

fhunleth replied:
Just because something has been merged, doesn't mean it works. There were trivial fixes yesterday, so has anyone done a delta update with NervesHub. Also, did it save 95%? Also, the doc PR hasn't been merged.

coryjewell replied:
Alright I'll change it to say coming soon

fhunleth replied:
Thanks!!!

fhunleth reviewed (approved):
Aside from the comment about delta updates, the new site looks good based on skimming the strings. I'm curious what it looks like for real, but I assume others will be pickier on that part, so I'm just going to mark this as approved by me.

jjcarstens commented on apps/nerves_hub_www/lib/nerves_hub_www_web/templates/home/index.html.eex:
```diff
@@ -1,39 +1,148 @@
+<div class="home-about">
+  <div class="content-container">
+    <div class="section-header josh-js" data-josh-anim-name="fadeIn">
+      <h2 class="ta-c">Archieve IoT agility with NervesHub</h2>
```
```suggestion
      <h2 class="ta-c">Achieve IoT agility with NervesHub</h2>
```

coryjewell replied:
whoops!

jjcarstens commented on apps/nerves_hub_www/lib/nerves_hub_www_web/templates/home/index.html.eex:
```diff
@@ -1,39 +1,148 @@
+      <h4>Manage Devices with Remote Console Monitoring</h4>
+      <p>Interact with your device remotely and securely through an intuitively designed console. See device connection status and health, recent activity and deployments, current firmware versions, and more.</p>
+    </div>
+    <img src="/images/home/manage-devices.png" alt="Manage devices" />
```
<img width="1135" alt="image" src="https://user-images.githubusercontent.com/11321326/94277726-30fd9300-ff07-11ea-97d3-6798f01b4f75.png">

So this seems to be the same image that's used in all these sections. I feel like this would prob be a good place to show a blip of the remote console? (Maybe that is planned and coming?)

coryjewell replied:
Making and image for this right now

coryjewell replied:
Updated image, nice suggestion!

jjcarstens commented on apps/nerves_hub_www/lib/nerves_hub_www_web/templates/home/index.html.eex:
```diff
@@ -1,39 +1,148 @@
+  <h2 class="ta-c"><span class="color-primary">Nerves</span> in Action</h2>
+  <div class="grid">
+    <div class="item">
+      <img src="/images/home/le%20tote.webp" alt="le tote">
```
I see these images exist in the PR, but I can't seem to get to them to work locally. Just a setup problem for me?

<img width="1334" alt="image" src="https://user-images.githubusercontent.com/11321326/94279829-e4678700-ff09-11ea-99eb-1e616bf7cd1f.png">

coryjewell replied:
I'm going to update these to be .jpg files instead of .webp

jjcarstens reviewed (commented):
Just a few things. One spelling error. Otherwise, 🎉 🙌

jjcarstens commented:
O..also..Is there a "dark mode" option for this homepage? It seems odd the site is all dark except for the homepage?

coryjewell commented:
There's not a dark mode as of now. The designs for this page called for this version. I'll actually play around with this locally today to check out a dark version instead

jjcarstens reviewed (approved):
:shipit:

mobileoverlord commented:
🎉
